man/man1/stty.1
.pa 1
.he 'STTY (I)'6/12/72'STTY (I)'
.ti 0
NAME		stty -- set teletype options
.sp
.ti 0
SYNOPSIS	stty____ option918 ...
.sp
.ti 0
DESCRIPTION	Stty____
will set certain I/O options on the current output teletype.
The option strings are
selected from the following set:

.in +3
.nf
even____	allow even parity.
-even_____	disallow even parity.
odd___	allow odd parity
-odd____	disallow odd parity
raw___	raw mode input
	(no erase/kill/interrupt/quit/EOT)
-raw____	negate raw mode
-nl___	allow cr for lf (and echo lf cr)
nl__	allow nl only
echo____	echo back every character typed.
-echo_____	do not echo characters.
lcase_____	map upper case to lower case
-lcase______	do not map case
-tabs_____	replace tabs by spaces
tabs____	preserve tabs
delay_____	calculate cr and tab delays.
-delay______	no cr/tab delays
ebcdic______	ebcdic ball conversion (2741 only)
corres______	correspondence ball conversion (2741 only)
.in -3
.sp
.fi
.ti 0
FILES		standard output.
.sp
.ti 0
SEE ALSO	stty(II)
.sp
.ti 0
DIAGNOSTICS	"Bad options"
.sp
.ti 0
BUGS		--


named/CHANGES
This is version 4.8.1 of bind.  It is a test release that includes
version 4.8 with fixes, asynchronous zone transfer and better reload
capabilities.  Although it is currently described as a test release,
it is believed to be reasonably stable and more usable than the currently-
released version, 4.8.  The changes of note are:

  o The asynchronous zone transfer code previously posted to the bind
    mailing list has been integrated, completed and tested.  There are
    a number of changes from the version posted, including fixes to
    allow top-level domains to work and a simplification of the timer
    code.

  o The code for reloading the server has been changed so that only
    primary zones master files that have changed are reloaded.  The
    cache and secondary zones are not flushed, and the sortlist, domain,
    etc. are reset to correspond to the boot file contents.

  o Several bugs have been fixed: the name "*" is not interpreted as
    a wildcard in cached zones, only in primary zones.  Secondary servers
    no longer decrement the time-to-live of records by the time since
    they verified the zone with the master; as a result, they never
    hand out nameserver referrals with too short a TTL to be usable.
    A bug was fixed that caused secondary servers with out-of-date
    zones to return empty answers between the actual expiration time
    and the next timeout.

There are several other bugs that have been reported but have not yet
been fixed.  In addition, the next regular release of named will
support negative caching, but this has not been integrated.

I would appreciate receiving feedback on this release; in particular,
problems (or lack of problems) when installing on various systems.
I attempted to update the SysV code when integrating the zone-transfer,
but haven't tested it.

The notes from version 4.8 follow.

----------
Welcome to version 4.8 of bind.

There have been several changes to the named boot file (/etc/named.boot)
of which you should be aware.  The "domain" line for each zone is no longer
needed, but one such line may still be used to specify a default domain
to be used for queries containing names with only a single component.
The term "suffixes", which was added in version 4.7alpha, has been removed.

The manual page on named (named.8) has been updated to reflect all
these changes.  Please read this and look at the example files
before installation.  You should also note the changes in the
resolver code to support non-fully-qualified addresses and per-user
host aliases.  See hostname(7) for an overview.  Two new routines
have been added to the resolver library since the last test release:
res_query formulates a query, sends it, waits for a response and does
preliminary error checking; res_search implements the search rules
of gethostbyname using res_query.

The MX lookup routine in sendmail has been modified to use res_search.  
Also, dn_skip takes an additional parameter and has been renamed
to dn_skipname.  While old sendmail binaries will work with the new
version of bind, because of these changes, it is desirable to install
new sendmail sources and recompile sendmail.  Do not rebuild sendmail 
from old sources.   The new sendmail is on ucbarpa.Berkeley.EDU for
anonymous FTP from pub/4.3/sendmail.MX.tar and pub/4.3/sendmail.MX.tar.Z.

There have been numerous changes to named, fixing most of the known
bugs that can be fixed without major structural changes in the server.
Several server configurations that failed before should now work.
Certain robustness problems have been fixed, in particular bounds-
checking when processing incoming packets.  Two changes have been made
in preparation for negative caching: SOA records are sent in the authority
section in negative responses with NXDOMAIN set, and a bug was fixed that
caused confusion and repeated requests if a response had no error, no answer
and an SOA in the authority section.  As such responses are already sent
by other servers, and will be sent by the next release of BIND, it is
important that all sites upgrade to this version as quickly as possible.

The root "hint" cache and cache file remain the largest problem area,
along with named's naivete in accepting bogus server's data.
These will be addressed in the next release, along with asynchronous
zone transfers, intelligent reloading of zone files, faster startup,
and caching of negative responses.

This version (4.8) will replace the last officially released version (4.5).
Version 4.5 has a serious bug that causes the generation of a continuous
stream of bogons to the root domain servers (bogus queries with the query
response bit set and possibly garbage for nsid and rcode).  It is imperative
that these versions of named be replaced as fast as possible.  We urge you to
field 4.8 quickly, for the sake of the root domain servers.

	Mike Karels
	Jean Wood
	bind@ucbarpa.Berkeley.EDU


usr/source/s2/stty.c
char	*arg;
int	mode[3];

main(argc, argv)
char	*argv[];
{
	int i;

	gtty(1, mode);
	while(--argc > 0) {

		arg = *++argv;
		if(eq("erase") || eq("kill") || eq("ek"))
			mode[1] = '#@';
		if(eq("even"))
			set(0200);
		if(eq("-even"))
			reset(0200);
		if(eq("odd"))
			set(0100);
		if(eq("-odd"))
			reset(0100);
		if(eq("raw"))
			set(040);
		if(eq("-raw") || eq("cooked"))
			reset(040);
		if(eq("-nl"))
			set(020);
		if(eq("nl"))
			reset(020);
		if(eq("echo"))
			set(010);
		if(eq("-echo"))
			reset(010);
		if(eq("lcase"))
			set(04);
		if(eq("-lcase"))
			reset(04);
		if(eq("-tabs"))
			set(02);
		if(eq("tabs"))
			reset(02);
		if(eq("-delay"))
			set(01);
		if(eq("delay"))
			reset(01);
		if(eq("hup"))
			set(0400);
		if(eq("-hup"))
			reset(0400);
		if(eq("tdelay"))
			reset(010000);
		if(eq("-tdelay"))
			set(010000);
		if(arg)
			printf("unknown mode: %s\n", arg);
	}
	stty(1,mode);
}

eq(string)
char *string;
{
	int i;

	if(!arg)
		return(0);
	i = 0;
loop:
	if(arg[i] != string[i])
		return(0);
	if(arg[i++] != '\0')
		goto loop;
	arg = 0;
	return(1);
}

set(b)
{

	mode[2] =| b;
}

reset(b)
{

	mode[2] =& ~b;
}
